cluster.go

```diff
@@ -1069,6 +1075,15 @@ func (r *connretrycache) ResetLen(n int) {
 	}
 }
 
+func (g group) pickReplica() string {
+	// no replica exists
+	if len(g.nodes) <= 1 {
+		return ""
```

Comment:
Hi @NeoHuang,

Almost there. I find that this line is not covered by tests:
<img width="522" alt="image" src="https://github.com/redis/rueidis/assets/2727535/2d68d251-2df3-40c1-848f-2ffc2dc01f46">

However, I am also wondering if the double-checking logic `len(g.nodes) <= 1` and `replicaAddr != ""` could be merged?

For example just:
```go
addr := master
if c.opt.ReplicaOnly && len(g.nodes) > 1 {
    addr = g.nodes[1+rand.Intn(len(g.nodes)-1)]
}
```